rhapsody/heatmap.py

```diff
@@ -105,6 +106,96 @@ def ranks_heatmap(ranks, microbe_metadata=None, metabolite_metadata=None,
     return hotmap
 
 
+def paired_heatmaps(ranks, microbes_table, metabolites_table, microbe_metadata,
+                    features, level, normalize='log10', color_palette='magma'):
+    '''
+    ranks: pd.DataFrame of conditional probabilities.
+        Microbes (rows) X metabolites (columns).
+    microbes_table: biom.Table
+        Microbe feature abundances per sample.
+    metabolites_table: biom.Table
+        Metabolite feature abundances per sample.
+    microbe_metadata: pd.Series
+        Microbe metadata for annotating plots
+    features: list
+        Select microbial feature IDs to display on paired heatmap.
+    level: int
+        taxonomic level for annotating clustermap.
+        Set to -1 if not parsing semicolon-delimited taxonomies.
+    normalize: str
+        Column normalization strategy to use for heatmaps. Must
+        be "log10", "z_score", or None
+    color_palette: str
+        Color palette for heatmaps.
+    '''
+    # validate microbes
+    missing_microbes = set(features) - set(microbes_table.ids('observation'))
+    if len(missing_microbes) > 0:
+        raise ValueError('features must represent feature IDs in '
+                         'microbes_table. Missing microbe(s): {0}'.format(
+                            missing_microbes))
+
+    # filter select microbes from microbe table and sort by abundance
+    sort_orders = [False] + [True] * (len(features) - 1)
+    select_microbes = microbes_table.copy().filter(features, 'observation')
+    select_microbes = select_microbes.to_dataframe().T
+    select_microbes = select_microbes[features]  # sort cols by features order
+    select_microbes = select_microbes.sort_values(
+        features[::-1], ascending=sort_orders)
+
+    # find top 50 metabolites (highest positive ranks)
+    microb_ranks = ranks.loc[features]
+    top_metabolites = microb_ranks.max()
+    top_metabolites = top_metabolites.sort_values(ascending=False)[:50].index
```

Comment:
```suggestion
    top_metabolites = top_metabolites.sort_values(ascending=False)[:top_k_metabolites].index
```

We'll want to have top_k_metabolites as a user-specified feature